### PR TITLE
OSX compatibility changes

### PR DESCRIPTION
**Pull request by Be3y4uu-K0T**
Pull request with regard to issue: #12

### CODE REVIEW

File: include/asyncio/selector/kqueue_selector.h [hunk 1]
```diff
@@ -13,35 +13,98 @@
 #include <cerrno>
 #include <cstdio>
 #include <vector>
+#include <chrono>
+#include <ranges>
+#include <memory>
+
+using namespace std::chrono;
```
Comment (netcan) [inline include/asyncio/selector/kqueue_selector.h:20]: `using namespace` declare in header is not recommend, just move this line to function block.

**Comment by Be3y4uu-K0T** [comment]
I don't understand. I tried to pass the check tests on my Ubuntu VM.
``` 
$ cat /etc/*release*
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.2 LTS"
PRETTY_NAME="Ubuntu 22.04.2 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.2 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
Codename:	jammy
$ git clone --recursive https://github.com/Be3y4uu-K0T/asyncio.git
$ cd asyncio
$ mkdir build-debug
$ cmake -H. -Bbuild-debug -DCMAKE_BUILD_TYPE=Debug
$ cmake --build build-debug -j2
...
[100%] Built target asyncio_ut
```
Everything works, but GitHub [tests](https://github.com/netcan/asyncio/actions/runs/4245017360/jobs/7388963233) don't pass with the same OS.
```
Operating System
  Ubuntu
  22.04.1
  LTS
```
What did I do wrong?

File: include/asyncio/stream.h [hunk 2]
```diff
@@ -9,12 +9,50 @@
...
+#include <sys/socket.h>
+#include <unistd.h>
 #include <netdb.h>
 #include <utility>
 #include <vector>
-#include <unistd.h>
+
+#if defined(__APPLE__) || defined(__CYGWIN__) || defined(__NetBSD__)
+    #include <sys/ioctl.h>
+    #include <fcntl.h>
```
Comment (netcan) [inline include/asyncio/stream.h:20]: `F_GETFL` defined in fcntl.h，but  ‘F_SETFL’ was not declared in this scope，move `#include <fcntl.h>` to the global include section?

**Comment by Be3y4uu-K0T** [comment]
I think the problem was in this line:
```c++
// file: <asyncio/stream.h>::34L
#if SOCK_NONBLOCK != 0
```
Due to the definition of the macro, the comparison was always false:
```c++
// file: <bits/socket_type.h>::52L
SOCK_NONBLOCK = 00004000      /* Atomically mark descriptor(s) as
                                   non-blocking.  */
#define SOCK_NONBLOCK SOCK_NONBLOCK
```
So I just replaced with if constexpr:
```c++
if constexpr (SOCK_NONBLOCK != 0) {
```

**Pull request merged**